src/librepl/repl.cpp

```diff
@@ -0,0 +1,191 @@
+ * If you have received a written license agreement or contract for        *
+ * Covered Software stating terms other than these, you may choose to use  *
+ * and redistribute Covered Software under those terms instead of these.   *
+ *                                                                         *
+ * Source is provided to this software because we believe users have a     *
+ * right to know exactly what a program is going to do before they run it. *
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include <librepl/librepl.h>
+#include <Python.h>
+
+#ifdef DRAKVUF_DEBUG
+
+extern bool verbose;
+
+#define PRINT_DEBUG(args...) \
+    do { \
+        if(verbose) fprintf (stderr, args); \
+    } while (0)
+
+#else
+#define PRINT_DEBUG(args...) \
+    do {} while(0)
+#endif
+
+#define Py_REF_DEBUG \
+    PyObject* refCount = PyObject_CallObject(PySys_GetObject("gettotalrefcount"), NULL); \
+    PRINT_DEBUG("total refcount = %i\n", PyInt_AsSsize_t(refCount)); \
+    Py_DECREF(refCount);
+
+static void repl_init(drakvuf_t drakvuf)
+{
+    // init python
+    Py_Initialize();
+
+    // load libdrakvuf
+    PyObject* sysPath = PySys_GetObject("path");
+    PyList_Append(sysPath, PyUnicode_FromString("/opt/drakvuf/src/librepl"));
```

Comment:
imo we should change this string, but the problem is that python needs absolute path here

Reply:
Make it configurable somehow?